### ISSUE TITLE
could not spawn `yum list nettle` on MacOS

### ISSUE THREAD

**Issue by stevengj** [staticfloat/Nettle.jl#74]
Doing `Pkg.build("Nettle")` with a fresh Julia master install gives:
```
LoadError: could not spawn `yum list nettle`: no such file or directory (ENOENT)
while loading /Users/stevenj/.julia/v0.6/Nettle/deps/build.jl, in expression starting on line 39
```
on MacOS, which is surprising since it shouldn't even be using  `yum` on a Mac.

Same error with both the tagged and master versions of Nettle, Homebrew, and BinDeps.

**Comment by stevengj** [comment]
Commenting out the `provides( Yum, "nettle", nettle )` line in deps/build.jl fixes the problem.

cc @tkelman, @Keno — has something changed recently in BinDeps that would break Nettle's build script?

**Comment by tkelman** [comment]
How recently? In theory the first page or two of https://github.com/JuliaLang/BinDeps.jl/commits/master is mostly deprecation fixes or test adjustments inside BinDeps. Can you bisect against BinDeps if it's repeatable?

**Comment by stevengj** [comment]
Yes, it's repeatable.  Git bisect says that the first bad commit is https://github.com/JuliaLang/BinDeps.jl/commit/91fce47c8f5f367b107aed242bb3979fbb7de4d7:
```
91fce47c8f5f367b107aed242bb3979fbb7de4d7 is the first bad commit
commit 91fce47c8f5f367b107aed242bb3979fbb7de4d7
Author: Ranjan Anantharaman <benditlikeranjan@gmail.com>
Date:   Fri Jul 22 16:04:47 2016 +0530

    Fix OS_NAME dep warn (#237)
    
    * Fix OS_NAME dep warn
    
    * Add fix
    
    Check if Unix, compare with KERNEL, if not, compare with :NT
    
    * Use OSNAME to denote the OS
    
    * Remove `@static` and fix line

:040000 040000 0c5546a47fc926c4fd083ecfd1b0bd4c0cd7dd50 cd8f24064674ee27908dfc8158ab99fe1eb392df M	src
```
cc @ranjanan

**Comment by stevengj** [comment]
(Which makes no sense to me.)

**Comment by tkelman** [comment]
maybe that does need to be `@static`?

**Comment by stevengj** [comment]
Wait, now I can no longer reproduce the problem... so the bisect results are probably meaningless.

**Comment by stevengj** [comment]
Grrr, I don't know what happened.  After bisecting, the problem suddenly went away.

**Comment by stevengj** [comment]
The problem seems to occur intermittently on Windows, too — see the AppVeyor failure in #75.

**Comment by tkelman** [comment]
Try on rc0? Both rc1 and the nightlies have some important pieces of the type system and precompilation a bit broken, unfortunately.

**Comment by yuyichao** [comment]
Should have been fixed on release-0.5 and master

**Comment by stevengj** [comment]
Awesome, I'll close this; will reopen if I ever see it again.
